internal/blocksync/pool.go

```diff
@@ -222,7 +223,7 @@ func (pool *BlockPool) PeekTwoBlocks() (first *types.Block, second *types.Block)
 }
 
 // PopRequest pops the first block at pool.height.
-// It must have been validated by 'second'.Commit from PeekTwoBlocks().
+// It must have been validated by 'second'.Commit from PeekTwoBlocks(), TODO: (?) and its corresponding ExtendedCommit.
```

Comment:
This will be changed with the verification in any case. The verification will include vote extensions where applicable.  But it is worth adding this comment here for reference.

Reply:
Yes, I'm aware that however strict we want to be with the current code, it won't be "enough" (as compared with the work @jmalicevic is doing).
That's why I'd propose to leave a comment if saying something like ("TODO: the first block should also be checked with the received extended commit")

Reply:
BTW, I have WIP  draft PR for the changes so we are aware of what is going on: https://github.com/tendermint/tendermint/pull/8419.